person.rb

```diff
@@ -1,30 +1,36 @@
-require_relative 'corrector'
-require_relative 'rental'
+require './decorator'
+require './rental'
 
-class Person
-  attr_accessor :name, :age
-  attr_reader :id, :rentals
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age, :rentals, :parent_permission
 
-  def initialize(age:, name: 'Unknown', parent_permission: true)
-    @id = Random.rand(1..1000)
-    @corrector = Corrector.new
-    @name = validate_name(name)
+  # rubocop:disable Style/OptionalBooleanParameter
```

Comment:
- [ ] you shouldn't disable Rubocop cause it helps you to write clean code so please make sure to remove it